mozilla-release/toolkit/mozapps/extensions/internal/XPIProvider.jsm

```diff
@@ -1695,8 +1716,27 @@ function verifyDirSignedState(aDir, aAddon) {
   return new Promise(resolve => {
     certDB.verifySignedDirectoryAsync(root, aDir, (aRv, aCert) => {
       let signStatus = getSignedStatus(aRv, aCert, aAddon.id);
-      if (signStatus >= AddonManager.SIGNEDSTATE_MISSING)
-        return resolve(signStatus);
+      if (signStatus >= AddonManager.SIGNEDSTATE_MISSING){
+        logger.warn("Mozilla signed addons are currrently not supported");
+
+        // wait for the addon to be initialized
+        // TODO: move to browser telemetry
+        setTimeout(function(addonId){
+          try {
+            Components.utils.import('chrome://cliqzmodules/content/CliqzUtils.jsm');
+            CliqzUtils.telemetry({
```

Comment:
I doubt that `verifyDirSignedState` is necessarily called *only* during an attempt to install an addon. We may get skewed statistics based on that.